dao.py

```diff
@@ -0,0 +1,49 @@
+from flask_pymongo import PyMongo
+from bson.objectid import ObjectId
+
+from app import app
+
+DB_NAME = 'schedule_practice'
+DB_USERNAME = 'admin'
+DB_PASSWORD = 'admin'
+DB_URI = 'mongodb://%s:%s@ds239387.mlab.com:39387/schedule_practice' % (DB_USERNAME, DB_PASSWORD)
+
+class DAO:
+    app.config['MONGO_DBNAME'] = DB_NAME
+    app.config['MONGO_URI'] = DB_URI
+    __mongo = PyMongo(app)
+
+    """
+    Read Operations
+    """
+    def _get_by_id(self, collection, id_str:str):
+        return self.__mongo.db[collection].find_one({'_id': ObjectId(id_str)})
+
+    def _get_by_params(self, collection, selector):
```

Comment:
`get_by_params` should implement `find_many` not `find_one` by accepting a collection of values to query by